src/main/java/ai/chat/repository/UserRepository.java

```diff
@@ -3,7 +3,9 @@
 import ai.chat.model.User;
 import org.springframework.data.mongodb.repository.MongoRepository;
 
+import java.util.Optional;
+
 public interface UserRepository extends MongoRepository<User, String> {
-  User findByUsername(String username);
```

Comment:
No need to remove this method, as we use this when creating a new user. Let's keep both methods findByUsername and findById.